Check for valid HTML5

ferki opened the pull request:
This PR enables HTML5 linting of the generated pages, and fails hard if there are any problems.

The built-in linter plugin simply prints warnings, but otherwise continues happily. In order to make it more useful with our Travis pipeline, I decided to simply import the standard plugin and patched it to fail on errors instead.

This patch could be made optional, and then it might be a good candidate to send upstream. If there's a better/simpler approach to override the behavior, don't hesitate to let me know :)

ehuelsmann commented on lib/Statocles/Plugin/HTMLLint/RexOps.pm:
```diff
@@ -0,0 +1,87 @@
+=attr plugins
+
+The L<HTML::Lint::Pluggable> plugins to use. Defaults to a generic set of
+plugins good for HTML5: 'HTML5' and 'TinyEntitesEscapeRule'
```
`TinyEntitesEscapeRule` instead of `TinyEntitiesEscapeRule` ?

ferki replied:
Yep, noticed that too, but seems like valid: https://metacpan.org/pod/release/KARUPA/HTML-Lint-Pluggable-0.10/lib/HTML/Lint/Pluggable/TinyEntitesEscapeRule.pm

ehuelsmann replied:
ouch. Agreed.

ehuelsmann commented on cpanfile:
```diff
@@ -1,4 +1,5 @@
 requires 'HTML::Entities';
+requires 'HTML::Lint::Pluggable';
```
`HTML::Lint::Pluggable` depends on `HTML::Lint` and this is a quote from the latest release notes of HTML::Lint:

> Note that this very well may be the final release of HTML::Lint that I make. I've been spending my time on HTML::Tidy5, which works on HTML5, checks for more problems, and is much faster.

Should we consider HTML::Tidy5 instead?

ferki replied:
I did, but it relies on an [external dependency](https://metacpan.org/pod/HTML::Tidy5#INSTALLING-TIDY).

Since the `HTML::Lint::Pluggable` still works, and apparently can catch some real problems, I decided to go for it. It also only added an extra second or so to lint all pages, so I am fine with that speed.

ehuelsmann replied:
ok. lets accept that then.

ehuelsmann commented on site.yml:
```diff
@@ -17,6 +17,12 @@ site:
         plugins:
             highlight:
                 $class: Statocles::Plugin::Highlight::RexOps
+            lint:
```
I know the synopsis and this section are semantically equivalent, but how about using *exactly* the same content instead?

ferki replied:
You mean updating the synopsis in the RexOps linter module?

ehuelsmann replied:
yes, I mean updating the synopsis.

ferki replied:
Fixed in followup commit.

ferki commented:
@ehuelsmann: I've added a few followup commits, so it's easier to review the additional changes. I will squash them away before merging.

I addressed your remaining feedback item, and decided to add a `fatal` attribute to the plugin control calling `die()` at the end or not.

ehuelsmann commented:
Very nice improvement! I love how the site becomes better every monh!

ehuelsmann commented:
oh. dang. you wanted to squash them!

ferki commented:
Since it's still very fresh, and it's probably not used by anyone else, I decided to quickly squash on master after discussing it with @ehuelsmann.

If you already pulled, you might need to reset your local master branch to the current one. In this case, sorry for the inconvenience!

To lower the likelihood of simiIar cases in the future, I'll use the autosquashable commits only in draft PRs, and I'll squash directly on mergeable PRs if needed.
